plugins/org.eclipse.thym.core/src/org/eclipse/thym/core/engine/HybridMobileEngineManager.java

```diff
@@ -99,14 +116,105 @@ public HybridMobileEngineManager(HybridProject project){
 	private boolean engineMatches(Engine configEngine, HybridMobileEngine engine){
 		//null checks needed: sometimes we encounter engines without a name or version attribute.
 		if(engine.isManaged()){
-			return configEngine.getName() != null && configEngine.getName().equals(engine.getId()) &&
-				configEngine.getSpec() != null && configEngine.getSpec().equals(engine.getVersion());
+
+			// Since cordova uses semver, version numbers in config.xml can begin with '~' or '^'.
+			if (configEngine.getSpec() != null) {
+				String spec = configEngine.getSpec();
+				if (spec.startsWith("~") || spec.startsWith("^")) {
+					spec = spec.substring(1);
+				}
+				return configEngine.getName() != null && configEngine.getName().equals(engine.getId())
+						&& spec.equals(engine.getVersion());
+			} else {
+				return false;
+			}
 		}else{
 			return engine.getLocation().isValidPath(configEngine.getSpec()) 
 					&& engine.getLocation().equals(new Path(configEngine.getSpec()));
 		}
 	}
-	
+
+	/**
+	 * Returns the active engines for the project by looking at
+	 * the values stored in platforms.json.
+	 *
+	 * </p>
+	 * If no engines are found in platforms.json, returns an empty array.
+	 * The file platforms.json is where the currently active cordova engines
+	 * are stored, (semi-)independently of what is stored in config.xml.
+	 *
+	 * @see HybridMobileEngineManager#defaultEngines()
+	 * @see HybridMobileEngineManager#getActiveEngines()
+	 * @return possibly empty array of {@link HybridMobileEngine}s
+	 */
+	public HybridMobileEngine[] getActiveEnginesFromPlatformsJson(){
+		try {
+			IFile file = project.getPlatformsJSONFile();
+			if (file == null) {
+				return new HybridMobileEngine[0];
+			}
+
+			List<HybridMobileEngine> activeEngines = new ArrayList<HybridMobileEngine>();
+
+			JsonParser parser = new JsonParser();
+			JsonObject root = parser.parse(new InputStreamReader(file.getContents())).getAsJsonObject();
+			for (String platform : PlatformConstants.SUPPORTED_PLATFORMS) {
+				if (root.has(platform)) {
+					HybridMobileEngine engine =
+							getHybridMobileEngine(platform, root.get(platform).getAsString());
+					if (engine != null) {
+						activeEngines.add(engine);
+					}
+				}
+			}
+			return activeEngines.toArray(new HybridMobileEngine[activeEngines.size()]);
+
+		} catch (JsonIOException e) {
+			HybridCore.log(IStatus.WARNING, "Error reading input stream from platforms.json", e);
+		} catch (JsonSyntaxException e) {
```

Comment:
Currently, Json errors have to be fixed manually. The file is simple enough that this shouldn't be a significant problem.